File: Back/models/product.js
import mongoose from "mongoose";
import Pharmacy from "./pharmacy.js";
const {Schema, model} = mongoose

const productSchema = new Schema(
    {
        name: {
            type: String,
            required: true
        },
        image: {
            type: String,
            required: true
        },
        price: {
            type: String,
            required: true
        },
        pharmacy: {
            type: Pharmacy,
            required: true
        },
    },
    {
        timestamps: true
    }
);

export default model('Product', productSchema);

File: Front/app/src/main/java/com/AndroidProject/app/appcomponents/models/Product.kt
package com.androidproject.app.appcomponents.models

import com.google.gson.annotations.SerializedName
import java.io.Serializable

data class Product(
    @SerializedName("_id")
    var id: String? = null,
    var name: String? = null,
    var image: String? = null,
    var price: String? = null,
    var pharmacy: List<Pharmacy>? = null,
    ): Serializable
